Framework/API/src/Algorithm.cpp

```diff
@@ -434,17 +434,10 @@ void Algorithm::unlockWorkspaces() {
 }
 
 //---------------------------------------------------------------------------------------------
-/** The actions to be performed by the algorithm on a dataset. This method is
- *  invoked for top level algorithms by the application manager.
- *  This method invokes exec() method.
- *  For Child Algorithms either the execute() method or exec() method
- *  must be EXPLICITLY invoked by the parent algorithm.
- *
- *  @throw runtime_error Thrown if algorithm or Child Algorithm cannot be
- *executed
- *  @return true if executed successfully.
+/** Invoced internally in execute()
  */
-bool Algorithm::execute() {
```

Comment:
I am wondering if you could have simply added the `AlgoTimeRegister::Dump` using a pre-processing directive inside `Algorithm.cpp` instead of having to add a new file and move the call to `execute()`. Maybe it would be less confusing?

Reply:
The reason is that I don't want to clutter up the source code with macroses especially because we need to have the spetial solutions for different OSes. The similar things in mantid have done in a this way.